src/FakeItEasy.Analyzer/UnusedReturnValueAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using static DiagnosticDefinitions;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class UnusedReturnValueAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly ImmutableDictionary<string, DiagnosticDescriptor> Diagnostics =
+            GetDiagnosticsMap(
+                nameof(UnusedCallSpecification));

```

Comment:
I assumed the description could be generated at need by keying off the attribute constructor argument. I did not think it totally through.